src/Swarm/TUI/Model/Achievement/Persistence.hs

```diff
@@ -4,38 +4,73 @@ module Swarm.TUI.Model.Achievement.Persistence where
 
 import Control.Arrow (left)
 import Control.Carrier.Lift (sendIO)
+import Control.Monad (forM, forM_)
+import Data.Either (partitionEithers)
 import Data.Text (Text, pack)
+import Data.Text qualified as T
 import Data.Yaml qualified as Y
 import Data.Yaml.Aeson (prettyPrintParseException)
 import Swarm.TUI.Model.Achievement.Attainment
 import Swarm.Util
 import System.Directory (
+  createDirectoryIfMissing,
+  doesDirectoryExist,
   doesFileExist,
+  listDirectory,
  )
 import System.FilePath
+import System.IO (hPutStrLn, stderr)
 
 -- | Get path to swarm achievements, optionally creating necessary
 --   directories.
 getSwarmAchievementsPath :: Bool -> IO FilePath
-getSwarmAchievementsPath createDirs =
-  (</> "achievements") <$> getSwarmDataPath createDirs
+getSwarmAchievementsPath createDirs = do
+  dp <- getSwarmDataPath False
+  let achievementsDir = dp </> "achievement"
```

Comment:
I changed this from `achievements` to `achievement` so that any recent players of the game don't get an error due to that path not being a directory.